pointcloud/basic_octree.go

```diff
@@ -36,6 +36,7 @@ type basicOctreeNode struct {
 	nodeType NodeType
 	children []*BasicOctree
 	point    PointAndData
+	maxVal   float64
```

Comment:
[q] Should we keep this as an int since that's what the Data.Value() type is?

Reply:
Ignore, I see that math.NaN is a flaot64

Reply:
we can't use `math.NaN` we want `int`

Reply:
changed. Since `int(math.NaN()) == 0` for emptyProb, we now do: `int(math.Inf(-1))`

Reply:
@EshaMaharishi @spieswl final thoughts on the benefits of using `math.NaN` being able to call `math.IsNaN()` to evaluate for data's presence versus `math.Inf(-1)` and being able to correctly represent `d.Value`'s type which is an int. 

Both work for this purpose, and are almost identical in implementation, although the NaN approach does have two addition small if statements in the octree code.

Reply:
Ah, that's unfortunate that `math.NaN` is a float, and we are storing int's.

That's odd, in Go Playground I am able to cast `math.NaN` to an int, and it prints as the same value as casting `math.Inf(-1)` to an int: https://go.dev/play/p/US14hvkeM5E?v=goprev

If that's the case, I think `math.NaN` would be better because it will be easier to cache other things besides maximum's in the future (e.g., averages). But I don't feel strongly and happy with whatever you all prefer.

Reply:
I'm pretty against casting a "not a number" to an int and getting a number out. It's not obvious exactly what the expected behavior should be, case in point this comment thread.

If we stick with `int`, consider explicitly using `math.MinInt` rather than implicitly generating it by casting floats that mean something else.

Reply:
Casting infinite or NaN to int is undefined behavior according to the Go spec and cannot be relied upon to always yield the same value.

> In all non-constant conversions involving floating-point or complex values, if the result type cannot represent the value the conversion succeeds but the result value is implementation-dependent.